lib/tictactoe.rb

```diff
@@ -1,42 +1,78 @@
+require 'ai_basic'
+require 'ai_minimax'
 require 'game_board'
 require 'game_rules'
+require 'game_state'
 require 'player_manager'
-require 'game_loop'
-require 'game_screen'
-class TicTacToe
 
-  attr_accessor :board, :game_loop
+class TicTacToe
 
-  def initialize (game_io, player1, player2)
+  def initialize
```

Comment:
This change severely impacts the testability of this class—instead of passing in dependencies they are now hardcoded. I don't think that should be necessary. What I _suspect_ was going on here (correct me if I'm wrong) is you were thinking any application that ends up using this gem shouldn't need to know too many details about how to wire things up together. There is some truth to that, but I think you may have gone a little overboard. Turning the core TTT logic into a gem shouldn't require abandoning all the nice design patterns you originally implemented when it was a full application.